arcgis-ios-sdk-samples/Maps/Download preplanned map/MapSelectionTableViewController.swift

```diff
@@ -194,23 +194,30 @@ class MapSelectionTableViewController: UITableViewController {
             offlineTask = AGSOfflineMapTask(onlineMap: onlineMap)
         }
         
-        guard let task = offlineTask else { return }
-        
+        let task = offlineTask!
         try? FileManager.default.removeItem(at: area.mapPackageURL)
-        
-        let job = task.downloadPreplannedOfflineMapJob(with: area, downloadDirectory: area.mapPackageURL)
-        currentJobs[area] = job
-        
-        let cell = tableView.cellForRow(at: indexPath) as? PreplannedMapAreaTableViewCell
-        cell?.progressView.observedProgress = job.progress
-        
-        job.start(statusHandler: nil) { [weak self] (result, error) in
+        let cell = self.tableView.cellForRow(at: indexPath) as? PreplannedMapAreaTableViewCell
```

Comment:
Please move this to inside the completion handler. Since cells are reused, this cell may or may not be the cell we want when `task.defaultDownloadPreplannedOfflineMapParameters` completes. That is why the method takes an index path, not a cell.

Reply:
Thanks for pointing that out, I fixed it.  
I don't fully understand the mechanism of `dequeueReusableCell` though. 🤔  Previously what I did is to refer to a cell first, and update it async afterwards, in this case in an escaping closure. If that cell that I referred to moves out of the view before the completion closure get executed, will it be reused and put into the queue, which cause the problem (not the same cell) you mentioned? 

Am I understanding it correctly or there is another reason that the cell might change? Thanks!

Reply:
That's correct. If a cell is scrolled offscreen, it is put back into the reuse queue and most likely dequeued for use by a different row. Retrieving the cell before performing an asynchronous operation and then updating it afterwards could lead to updating the cell for the wrong row, causing both the expected row and the accidentally modified row to be incorrect.